includes/class-freemius.php

```diff
@@ -3514,6 +3519,411 @@ private static function _load_required_static() {
             self::$_statics_loaded = true;
         }
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.4.3
+         *
+         * @return bool
+         */
+        private function is_unresolved_clone() {
+            if ( ! $this->is_clone() ) {
+                return false;
+            }
+
+            return ( ! FS_Clone_Manager::instance()->is_temporary_duplicate( false ) );
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.4.3
+         */
+        function is_clone() {
+            if ( ! is_object( $this->_site ) ) {
+                return false;
+            }
+
+            return ( trailingslashit( $this->_site->url ) !== trailingslashit( get_site_url() ) );
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.4.3
+         */
+        static function _init_clone_manager() {
+            if ( is_admin() ) {
+                FS_Clone_Manager::instance()->init();
+
+                if ( ! empty( FS_Clone_Manager::instance()->get_clone_identification_timestamp() ) ) {
+                    if ( self::is_ajax() ) {
+                        self::add_ajax_action_static( 'handle_clone_resolution', array( 'Freemius', '_clone_resolution_action_ajax_handler' ) );
+                    } else if ( ! self::is_cron() && ! self::is_admin_post() ) {
+                        self::maybe_show_clone_admin_notice();
+
+                        add_action( 'admin_footer', array( 'Freemius', '_add_clone_resolution_javascript' ) );
+                    }
+                }
+            }
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.4.3
+         */
+        static function _add_clone_resolution_javascript() {
+            $vars = array( 'ajax_action' => self::get_ajax_action_static( 'handle_clone_resolution' ) );
+
+            fs_require_once_template( 'clone-resolution-js.php', $vars );
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.4.3
+         */
+        static function _clone_resolution_action_ajax_handler() {
+            self::$_static_logger->entrance();
+
+            check_ajax_referer( self::get_ajax_action_static( 'handle_clone_resolution' ), 'security' );
+
+            $clone_action = fs_request_get( 'clone_action' );
+
+            if ( empty( $clone_action ) ) {
+                self::shoot_ajax_failure( array(
+                    'message'      => fs_text_inline( 'Invalid clone resolution action param.', 'invalid-clone-resolution-action-param-error' ),
+                    'redirect_url' => '',
+                ) );
+            }
+
+            if ( 'temporary_duplicate' === $clone_action ) {

```

Comment:
@fajardoleo clone actions should probably be constants in the clone manager instead of strings